### PR TITLE
Mobile - Rich Text - Validate link colors

### LINKED ISSUE #15876: JSApplicationIllegalArgumentException: Error while updating property 'text' of a view managed by: RCTAztecView
**Sentry Url**: https://sentry.io/share/issue/0b0f4f57de494b18ac35163cf4b7037a/
**User Count**: 21
**Count**: 109
**First Release**: 19.0
**First Seen**: 2022-01-28T16:48:22.458000Z
**Last Seen**: 2022-01-31T16:07:17Z
**24 Hours**: 40
**30 Days**: 112

```
JSApplicationIllegalArgumentException: Error while updating property 'text' of a view managed by: RCTAztecView

...
(17 additional frame(s) were not displayed)
```

@develric This seems very low in user count right now so probably low priority… but still wanted to report it here because it's new in 19.0, which got a very late rollout due to Google taking time approving it, so that might explain the low numbers. So probably worth keeping an eye on this one during the upcoming days to check if it evolves as 19.0 adoption grows.

### PR DESCRIPTION
**Pull request by geriux**
- `Gutenberg Mobile PR` -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/4542

Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/15876

## Description
This PR fixes a crash on production for users using block-based themes and custom link colors, currently we don't support custom palettes and this is generating setting invalid link colors. On iOS it just simply uses the default color, but on Aztec Android, if an invalid color is passed through the `text` attribute in `RichText` it will crash the app when it tries to parse it.

So to avoid this, it will validate if the color is correct, if not it'll pass the default editor value. Using the `colord` library to check the value.

### Precondition: A custom color palette, to create this follow these instructions:

- Select a block-based theme
- Enable Full site editing: On WordPress.com go to Settings > Full site editing (beta)
  <kbd>Screen Shot 2022-02-02 at 14 14 06</kbd>
- Go to Appearance > Editor (beta)
- On the upper right side select the Styles setting
  <kbd>Screen Shot 2022-02-02 at 14 16 12</kbd>
- Select the Colors option
- Open the Palette 
- Add a custom palette with a few colors
- Save the settings (you have to confirm saving)

### Test case 1 (Change the default link color of your theme to a custom one)

- Go to the site editor on WordPress.com
- On the right sidebar, under Styles (beta) go to Colors
   <kbd>Screen Shot 2022-02-02 at 14 19 14</kbd>
- Click on the Links color and select one from your **Custom** palette
- Save the settings (you have to confirm saving)
- Open the app with metro running (Sometimes you need to re-open the editor to load the theme data [36152](https://github.com/WordPress/gutenberg/issues/36152))
- Create a new post
- Add a Paragraph block
- Type in some text and add a link
- **Expect** to see the default link color and the app **not crashing**

**Note**: You won't see the link color you selected until we support custom palettes, this will be introduced [here](https://github.com/WordPress/gutenberg/pull/38417).

## Screenshots 

<kbd></kbd>

## Types of changes
Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [x] I've updated related schemas if appropriate.

### CODE REVIEW

File: packages/rich-text/src/component/index.native.js [hunk 1]
```diff
@@ -1051,6 +1061,9 @@ export class RichText extends Component {
 			textDecorationColor: defaultTextDecorationColor,
 			fontFamily: defaultFontFamily,
 		} = getStylesFromColorScheme( styles.richText, styles.richTextDark );
+		const textLinkColor = this.getLinkTextColor(
```
Comment (fluiddot) [inline packages/rich-text/src/component/index.native.js]: Nitpick: Rename the variable to match the `RCTAztecView` prop.
```suggestion
		const linkTextColor = this.getLinkTextColor(
```

**NOTE:** If this change is applied, keep in mind that it also requires updating line 1162.

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

I confirm that the crash can't be reproduced, however, I noticed that adding 3 or more colors to the custom palette makes the default text color turn white which makes the text to be unreadable (see attached video capture).

https://user-images.githubusercontent.com/14905380/152322366-b7d35e07-0344-4238-b6e7-ef50c2785672.mp4

File: packages/rich-text/src/component/index.native.js (hunk 1 above)
Comment (geriux) [reply packages/rich-text/src/component/index.native.js]: Thank you, updated in https://github.com/WordPress/gutenberg/pull/38474/commits/6b4d8ff1c5bae95539fed8cd2651f511d37decde

**Comment by geriux** [comment]
> I confirm that the crash can't be reproduced, however, I noticed that adding 3 or more colors to the custom palette makes the default text color turn white which makes the text to be unreadable (see attached video capture).

Yeah, that's one of the issues that will be fixed in https://github.com/WordPress/gutenberg/pull/38417. For now this will prevent the crash. Thanks for bringing it up and for the review!

**Comment by geriux** [comment]
The changelog update will be included in https://github.com/WordPress/gutenberg/pull/38488

**Pull request merged**